### PR TITLE
fix: chartWidget fix for "TypeError: r.palettecolors.split"

### LINKED ISSUE #7501: [Bug] TypeError: r.palettecolors.split is not a function
Sentry Issue: [APPSMITH-150](https://sentry.io/organizations/appsmith/issues/2650518421/?referrer=github_integration)

```
TypeError: r.palettecolors.split is not a function
  at componentDidUpdate (components/designSystems/appsmith/ChartComponent.tsx:476:28)
...
(45 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by rahulramesha**
## Description

fix sentry issue, for "TypeError: r.palettecolors.split"

Fixes #7501 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
Manual verification and jest

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix_chart_sentry 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 54.85 **(0.01)** | 36.72 **(0.06)** | 33.85 **(0.01)** | 55.46 **(0)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.47 **(-0.24)** | 38.26 **(-0.87)** | 36.21 **(0)** | 55.08 **(-0.27)**
 :green_circle: | app/client/src/workers/validations.ts | 75 **(0.88)** | 68.02 **(1.56)** | 65.63 **(2.3)** | 77.03 **(0.72)**</details>

### CODE REVIEW

File: app/client/src/workers/validations.ts [hunk 1]
```diff
@@ -304,7 +304,8 @@ export const VALIDATORS: Record<ValidationTypes, Validator> = {
     const isValid = isString(parsed);
     if (!isValid) {
       try {
-        parsed = toString(parsed);
+        if (!config.params?.strict) parsed = toString(parsed);
+        else throw Error("");
```
Comment (hetunandu) [inline app/client/src/workers/validations.ts:308]: Why is this error being thrown? How will it get resolved? Why no error message?

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/workers/validations.ts (hunk 1 above)
Comment (rahulramesha) [reply app/client/src/workers/validations.ts:309]: It is being caught in the next line, it is a pseudo error, which should help me avoid rewriting the same code to return  the object that is already being returned in the catch.

Comment (hetunandu) [reply app/client/src/workers/validations.ts:309]: Please extract the return object into a variable and return instead of throwing an error.

**Review by hetunandu — CHANGES_REQUESTED** [review]

**Review by hetunandu — APPROVED** [review]

**Review by riodeuno — DISMISSED** [review]

**Comment by vivekverma2312** [comment]
@rahulramesha Still able to repro the issue on Deploy Preview. Please check other errors also.
Screenshot 2021-09-23 at 3 06 55 PM

**Review by hetunandu — APPROVED** [review]

**Comment by vivekverma2312** [comment]
@rahulramesha Tested on deploy preview and working fine.

**Pull request merged**